<?xml version="1.0"?>
<robot xmlns:xacro="http://ros.org/wiki/xacro">
	<!--leg xacro-->
	<xacro:macro name="EISpider_Leg" params="LegName leg_x leg_y leg_yaw">
		<link name="link_${LegName}_1">
            <visual>
				<origin xyz="-0.065 0.080 -0.080" rpy="${PI/2} 0 0"/>
				<geometry>
					<mesh filename="package://hit_spider/model/meshes/leg1.stl"   scale="0.001 0.001 0.001"/>
				</geometry>
                <material name="xccc">
                    <color rgba="0.3 0.3 0.3 1"/>
                </material>
			</visual>
			<inertial>
				<origin xyz="${link1_length/2} 0 0" rpy="0 0 0"/>
				<mass value="${link1_mass}"/>
				<inertia
				ixx="${link1_ixx}" ixy="0.0" ixz="0.0"
				iyy="${link1_iyy}" iyz="0.0"
				izz="${link1_izz}"/>
			</inertial>
		</link>
		<link name="link_${LegName}_2">
			<visual>
				<origin xyz="-0.086 0.086 0.065" rpy="0 ${PI/2} ${-PI/2}"/>
				<geometry>
					<mesh filename="package://hit_spider/model/meshes/leg2.stl"   scale="0.001 0.001 0.001"/>
				</geometry>
                <material name="sssss">
					<color rgba="0.914 0.914 0.847 1"/>
                </material>
			</visual>

			<inertial>
				<origin xyz="${link2_length/2} 0 0" rpy="0 0 0"/>
				<mass value="${link2_mass}"/>
				<inertia
				ixx="${link2_ixx}" ixy="0.0" ixz="0.0"
				iyy="${link2_iyy}" iyz="0.0"
				izz="${link2_izz}"/>
			</inertial>
		</link>

		<link name="link_${LegName}_3">
			<visual>
				<origin xyz="0.373 -0.065 -0.085" rpy="0 0 ${PI/2}"/>
				<geometry>
					<mesh filename="package://hit_spider/model/meshes/leg3.stl"   scale="0.001 0.001 0.001"/>
				</geometry>
				 <material name="sssss"/>
			</visual>
			<visual>
				<origin xyz="0.40 -0.036 -0.036" rpy="0 0 0"/>
				<geometry>
					<mesh filename="package://hit_spider/model/meshes/foot.stl"   scale="0.001 0.001 0.001"/>
				</geometry>
			</visual>
			<collision>
				<origin xyz="${link3_length/2} 0 0" rpy="0 0 0"/>
				<geometry>
					<box size="${link3_length} 0.05 0.05"/>
				</geometry>
			</collision>
			<inertial>
				<origin xyz="${link3_length/2} 0 0" rpy="0 0 0"/>
				<mass value="${link3_mass}"/>
				<inertia
				ixx="${link3_ixx}" ixy="0.0" ixz="0.0"
				iyy="${link3_iyy}" iyz="0.0"
				izz="${link3_izz}"/>
			</inertial>
		</link>

		<link name="link_${LegName}_foot">
			<visual>
				<origin xyz="0 0 0" rpy="0 0 0"/>
				<geometry>
					<!--cylinder length="0.1" radius="0.05"/-->
					<sphere radius="0.03"/>
				</geometry>
			</visual>
			<collision>
				<origin xyz="0 0 0" rpy="0 0 0"/>
				<geometry>
					<!--cylinder length="0.1" radius="0.05"/-->
					<sphere radius="0.03"/>
				</geometry>
			</collision>
			<inertial>
				<origin xyz="0 0 0" rpy="0 0 0"/>
				<mass value="0.01"/>
				<inertia
				ixx="0.01" ixy="0.0" ixz="0.0"
				iyy="0.01" iyz="0.0"
				izz="0.01"/>
			</inertial>
		</link>

		<!--joints-->
		<joint name="joint_${LegName}_1" type="revolute">
			<parent link="link_trunk"/>
			<child link="link_${LegName}_1"/>
			<origin xyz="${leg_x} ${leg_y} 0" rpy="0 0 ${leg_yaw}"/>
			<axis xyz="0 0 1"/>
			<limit effort="350.0" lower="-${PI/2}" upper="${PI/2}" velocity="4.0"/>
			<dynamics damping="30"/>
		</joint>

        <joint name="joint_${LegName}_2" type="revolute">
			<parent link="link_${LegName}_1"/>
			<child link="link_${LegName}_2"/>
			<origin xyz="${link1_length} 0 0" rpy="${PI/2} 0 0"/>
			<axis xyz="0 0 1"/>
			<limit effort="500.0" lower="-${PI/3}" upper="${PI/3}" velocity="4.0"/>
			<dynamics damping="30"/>
		</joint>

		<joint name="joint_${LegName}_3" type="revolute">
			<parent link="link_${LegName}_2"/>
			<child link="link_${LegName}_3"/>
			<origin xyz="${link2_length} 0 0" rpy="0 0 ${-PI/2}"/>
			<axis xyz="0 0 1"/>
			<limit effort="500.0" lower="-${PI/2}" upper="${PI/2}" velocity="4.0"/>
			<dynamics damping="30"/>
		</joint>

		<joint name="joint_${LegName}_ft_sensor" type="fixed">
			<parent link="link_${LegName}_3"/>
			<child link="link_${LegName}_foot"/>
			<origin xyz="${link3_length} 0 0" rpy="0 ${PI/2} 0"/>
			<axis xyz="0 0 1"/>
		</joint>

		<xacro:include filename="$(find hit_spider)/model/xacro/leg/leg.transmission.xacro"/>
        <xacro:leg_transmission LegName="${LegName}"/>

        <xacro:include filename="$(find hit_spider)/model/xacro/leg/leg.gazebo.sensor.xacro"/>   
        <xacro:legFTsensor  LegName="${LegName}"/>
        
	</xacro:macro>
</robot>